File: generator/registry/playwright-python/files/pages/home_page.py
"""Example page object. Replace locators/methods with your application's.

Prefer role/text based locators (get_by_role, get_by_text) over brittle
XPath or nth-child CSS selectors.
"""
from playwright.sync_api import Page, Locator
from pages.base_page import BasePage


class HomePage(BasePage):
    def __init__(self, page: Page, base_url: str) -> None:
        super().__init__(page, base_url)

    def open(self) -> "HomePage":
        self.goto("/")
        return self

    def heading(self) -> Locator:
        return self.page.get_by_role("heading", level=1)

    def click_get_started(self) -> None:
        self.page.get_by_role("link", name="Get started").first.click()

File: generator/registry/playwright-ts/files/src/pages/HomePage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage.js';

/**
 * Example page object for the Playwright docs site (the default baseURL).
 * Replace the locators and methods with your own application's home page.
 *
 * Locator strategy: prefer role-based and getByText locators — they are
 * resilient to markup changes. Avoid brittle XPath / nth-child selectors.
 */
export class HomePage extends BasePage {
  private readonly getStartedLink: Locator;
  private readonly searchButton: Locator;
  private readonly heroTitle: Locator;

  constructor(page: Page) {
    super(page);
    this.getStartedLink = page.getByRole('link', { name: 'Get started' });
    this.searchButton = page.getByRole('button', { name: 'Search' });
    this.heroTitle = page.getByRole('heading', { level: 1 });
  }

  async open(): Promise<void> {
    await this.goto('/');
    await this.waitForLoad();
  }

  async clickGetStarted(): Promise<void> {
    await this.click(this.getStartedLink);
  }

  async openSearch(): Promise<void> {
    await this.click(this.searchButton);
  }

  heroHeading(): Locator {
    return this.heroTitle;
  }
}
